File: python/test/test_permutation.py
import ThrustRTC as trtc



dvalues = trtc.device_vector_from_list([10.0, 20.0, 30.0, 40.0, 50.0, 60.0, 70.0, 80.0], 'float')
dindices =  trtc.device_vector_from_list([2,6,1,3], 'int32_t')
doutput = trtc.device_vector('float', 4)

perm = trtc.DVPermutation(dvalues, dindices)

trtc.Transform(perm, doutput, trtc.Negate())
print (doutput.to_host())

File: cpp/test/test_permutation.cpp
#include <stdio.h>
#include "TRTCContext.h"
#include "DVVector.h"
#include "fake_vectors/DVPermutation.h"
#include "transform.h"

int main()
{
	float hvalues[8] = { 10.0f, 20.0f, 30.0f, 40.0f, 50.0f, 60.0f, 70.0f, 80.0f };
	DVVector dvalues("float", 8, hvalues);

	int hindices[4] = { 2,6,1,3 };
	DVVector dindices("int32_t", 4, hindices);

	float houtput[4];
	DVVector doutput("float", 4);

	DVPermutation perm(dvalues, dindices);

	TRTC_Transform(perm, doutput, Functor("Negate"));
	doutput.to_host(houtput);
	printf("%f %f %f %f\n", houtput[0], houtput[1], houtput[2], houtput[3]);

	return 0;
}
